geonode_mapstore_client/templatetags/get_menu_json.py

```diff
@@ -0,0 +1,38 @@
+from django import template
+from geonode.base.models import Menu, MenuItem
+
+register = template.Library()
+
+@register.simple_tag
+def get_menu_json(placeholder_name):
+    menus = {
+        m: MenuItem.objects.filter(menu=m).order_by('order')
+        for m in Menu.objects.filter(placeholder__name=placeholder_name)
+    }
+    ms = []
+    for menu, menu_items in menus.items():
+        if len(menu_items) > 1:
+            m = {}
+            m['label'] = menu.title
+            m['type'] = 'dropdown'
+            m['items'] = []
+            for menu_item in menu_items:
```

Comment:
Looks like that the handling of the single item is the same in both cases. A small refactor can put in common this lines of code, something like this:

```python
if len(menu_items) > 1:
......
    for menu_item in menu_items:
        m_item = handle_single_item(menu_item)
        m['items'].append(m_item)
```
Same for the single item:
```python
if len(menu_items) == 1:
    m = handle_single_item(menu_items.first())
    ms.append(m)
```
An the common function:
```python
def handle_single_item(menu_item):
    m_item = {}
    m_item['type'] = 'link'
    m_item['href'] = menu_item.url
    m_item['label'] = menu_item.title
    if menu_item.blank_target:
        m_item['target'] = '_blank'
    return m_item
```